### DISCUSSION TITLE
Repository code search breaks when search query includes '%'

### DISCUSSION THREAD

**Discussion by ewancg** [community/community#56654] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I went to search for a particular modulus expression in a repo and GitHub refused the GET request with error 400. This will happen with `%` anywhere in the search query parameter (`search?q=[%]`) as long as the repository is not empty. There may be other characters that are not properly escaped & will trigger such an error.
For example:
https://github.com/github/.github/search?q=%25; valid, working, redirects as expected
https://github.com/github/.github/search?q=%; blank screen (this is what Ctrl+K -> '%' -> Enter will do)

**Comment by look** [comment]
Thanks for reporting this. I'll notify the appropriate team.
